server/db/model/Team.js

```diff
@@ -0,0 +1,88 @@
+var mongoose = require('../index.js'),
+    sanitizerPlugin = require('mongoose-sanitizer-plugin'),
+    escapeStringRegex = require('escape-string-regexp');
+
+// Define the document Schema
+var schema = new mongoose.Schema({
+    name: {
+        type: String,
+        form: {
+            user_editable: true,
+            label: 'Team Name',
+            placeholder: 'Hacker McHackerTeam'
+        }
+    },
+    description: {
+        type: String,
+        form: {
+            user_editable: true,
+            label: 'Description',
+            placeholder: 'Your description goes here'
+        }
+    },
+    //First user in the array will be the 'leader'
+    members: [
+        {
+            type: mongoose.Schema.Types.ObjectId,
+            ref: 'User'
+        }
+    ]
+});
+
+//Checks the length of the description
+schema.path('description').validate(function(val) {
+    return val.length >= 100;
+}, 'A descrpition must be at least 100 characters.');
+
+// Allow us to query by name
+schema.query.byTeamName = function(name) {
+    var escapedName = escapeStringRegex(name);
+    return this.findOne({
+        name: new RegExp(escapedName, 'i')
+    });
+};
+
+schema.query.allMembers = function(teamId) {
```

Comment:
Somethings looks wrong here. Are you meaning to return in the callback?

Reply:
Yeah it was weird, I removed it because it's not necessary